tests/update_state_test.py

```diff
@@ -0,0 +1,202 @@
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is False
+
+        game.check_guess()
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
+        assert game.guess_value == 3
+        assert game.code_number == 0
+
+    def test_guessed_death_word(self):
+        'Test if flipped starts as false and ends as true.'
+        'Also check if guess is the death word and code_number is reset'
+        game = Game()
+        game.code_number = 3
+        # game.guess is the first word that is the death word
+        game.guess = [card.word for card in game.board.cards if card.color == "black"][0]
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is False
+
+        game.check_guess()
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
+        assert game.guess_value == 4
+        assert game.code_number == 0
+
+
+class TestCheckWin:
+    '''Check_win is responsible for assigning a winner if the death card 
+    is picked or if the entire deck has been flipped'''
+    def test_winner_and_game_over_for_death_card(self):
+        'test if winner_color is the opposite team and game_over is True'
+        game = Game()
+        game.guess_value = 4
+        game.team = "blue"
+
+        game.check_win()
+
+        assert game.winner_color == 'red'
+        assert game.game_over is True
+
+    def test_winner_for_death_card(self):
+        'test if when team is "red", winner_color is "blue"'
+        game = Game()
+        game.guess_value = 4
+        game.team = "red"
+
+        game.check_win()
+
+        assert game.winner_color == 'blue'
+        assert game.game_over is True
+
+    def test_red_complete_deck(self):
+        'test if when all red cards are flipped, winner_color = red and game_over is True'
+        game = Game()
+        # make all red cards guessed
+        for card in game.board.cards:
+            if card.color == 'red':
+                card.flipped = True
+
+        game.check_win()
+
+        assert game.winner_color == 'red'
+        assert game.game_over is True
+
+    def test_blue_complete_deck(self):
+        'test if when all blue cards are flipped, winner_color = blue and game_over is True'
+        game = Game()
+        # make all blue cards guessed
+        for card in game.board.cards:
+            if card.color == 'blue':
+                card.flipped = True
+
+        game.check_win()
+
+        assert game.winner_color == 'blue'
+        assert game.game_over is True
+
+class TestUpdateState:
+    '''Update_state sets guessing to True or False, then sets skip True first then False,
+    and finally, switches the team color and resets guess_value to zero'''
+    def test_guessing_is_true_under_right_conditions(self):
+        'If these three variables are as they are set here, guessing should be True'
```

Comment:
Possible clarification.

`'If our last guess was correct and we have guesses left and the game is not over, we should get another guess'`